language-server/modules/langserver-commons/src/main/java/org/ballerinalang/langserver/commons/workspace/WorkspaceManager.java

```diff
@@ -172,6 +172,15 @@ public interface WorkspaceManager {
      */
     Optional<PackageCompilation> waitAndGetPackageCompilation(Path filePath);
 
+    /**
+     * Returns module compilation from the file path provided.
+     *
+     * @param filePath file path of the document
+     * @param isSourceChange True if the given file's source is changed
+     * @return {@link ModuleCompilation}
+     */
+    Optional<PackageCompilation> waitAndGetPackageCompilation(Path filePath, boolean isSourceChange);
```

Comment:
@IMS94 do we need to change the boolean to LSOperation. If we do so, we'll need a default value for it. thoughts?

Reply:
Agreed. What if we move the common logic into a private method in the BallerinaWorkspaceManager class?

Reply:
Decided not to proceed with this. Hence resolving